File: src/Software Engineer Prepare Kit/17/solution.ts
const PHONE_DIGITS = ["0", "1", "abc", "def", "ghi", "jkl", "mno", "pqrs", "tuv", "wxyz"] as const

export default function minTasksToCancelForNoConflict(digits: string): string[] {
	if (!digits.length) return []

	const combinations: string[] = []

	const backtrack = (index: number, path: string[]) => {
		if (index === digits.length) {
			combinations.push(path.join(""))
			return
		}

		const digit = parseInt(digits[index]!)
		const letters = PHONE_DIGITS[digit]!

		for (const letter of letters) {
			path.push(letter)
			backtrack(index + 1, path)
			path.pop()
		}
	}

	backtrack(0, [])

	return combinations
}

File: src/Software Engineer Prepare Kit/17/solution.go
package main

var PHONE_DIGITS = [10]string{
	"0",
	"1",
	"abc",
	"def",
	"ghi",
	"jkl",
	"mno",
	"pqrs",
	"tuv",
	"wxyz",
}

func MinTasksToCancelForNoConflict(digits string) []string {
	if len(digits) == 0 {
		return []string{}
	}

	var combinations []string

	var backtrack func(index int, path []byte)

	backtrack = func(index int, path []byte) {
		if index == len(digits) {
			combinations = append(combinations, string(path))
			return
		}

		digit := digits[index] - '0'
		letters := PHONE_DIGITS[digit]

		for _, letter := range letters {
			path = append(path, byte(letter))
			backtrack(index+1, path)
			path = path[:len(path)-1]
		}
	}

	backtrack(0, make([]byte, 0, len(digits)))

	return combinations
}
